src/core/BaseService.ts

```diff
@@ -115,33 +119,20 @@ class BaseService {
     return this.delete<T>(id)
   }
 
-  submit<T = any>(requestType: Method): Promise<T>
-  submit<T = any>(requestType: Method, parameter?: string | number): Promise<T>
-  submit<T = any>(
-    requestType: Method,
-    parameter?: string | number,
-    form?: T,
-  ): Promise<T>
-  submit<T = any>(
-    requestType: Method,
-    parameter?: string | number,
-    form?: T,
-    config?: AxiosRequestConfig,
-  ): Promise<T>
   submit<T = any>(
```

Comment:
Getting rid of the overloads would interfere with custom usage of `submit` function. But it seemed easiest to do it like this for now. Can refactor it a bit. The PR is just about the idea, not the execution.